ledger/acctupdates.go

```diff
@@ -387,7 +406,13 @@ func (au *accountUpdates) lookupKv(rnd basics.Round, key string, synchronized bo
 
 		persistedData, err := au.accountsq.lookupKeyValue(key)
 		if persistedData.round == currentDbRound {
-			return persistedData.value, nil
+			if persistedData.value != nil {
+				// if we read actual data return it
+				au.baseBoxes.writePending(persistedData, key)
+				return persistedData.value, err
+			}
+			// otherwise return empty
+			return nil, err
 		}
 
 		// The db round is unexpected...
```

Comment:
@algoidurovic Leaving note on unrelated line because impacted line is _not_ in diff.

In `lookupKeysByPrefix`on line 526, it appears there's a placeholder to call the LRU cache.  Double checking: Does the PR deliberately make no change?

I spotted the concern by inspecting usage of `lruboxes.go::read` and considering other possible call sites.

Reply:
I deliberately didn't use the cache there because it isn't set up to make queries on box name prefixes. Also, `lookupKeysByPrefix` is only called by the API server afaik so it's probably not as important to speed up reads with a cache.

Reply:
I think this is the right decision.  We'd need a new datastructure to support range queries, and we'd have to think hard about cache coherence.  Let's skip it.  However, let's note that this is a point in favor of giving the caching job to sqlite (bigger cache? bigger pages?) as it would probably help there.

Reply:
@algoidurovic @jannotti I'm on-board with _not_ supporting the prefix use case.  

Prior to closing the thread, can an inline comment be added to `lookupKeysByPrefix` that explains _why_ there's no LRU cache usage?

Reply:
Added a comment.

Reply:
@algoidurovic Resolving - thanks for the inline comment.